Added Datadog plugin

bwebster opened the pull request:
## Overview
This plugin, when included, will attempt to use `seismograph` to report deploy events to Datadog.

To use this:
```
config.add_plugin(KapostDeploy::Plugins::NotifyDatadogAfterPromote)
```

## Notes
The plugin was written to be safe in situations where `seismograph` is not available by plugging in a null object notifier.

Also in this PR:
- Upgraded ruby version, so I could then upgrade the version of `gemsmith` we are using.  Any versions prior to the one I upgraded to have a dependency on `thor_plus`, which apparently has been yanked .
- Also updated our `rubocop` version due to a vulnerability in the current version we are using.

bwebster committed:
Updated ruby version to 2.4.2.

bwebster committed:
Updated gemsmith version due to thor_plus issue.

- Seems as though thor_plus was yanked and all versions of gemsmith
  prior to 10 rely on gemsmith, making it impossible to bundle install.

bwebster committed:
Added Datadog notification plugin.

- This plugin can be used to notify Datadog about deployments.  It
  will send timing information about how long the deploy took, as
  well as create a info event.

bwebster committed:
Upgraded rubocop version.

- This addresses a vulnerability in the previous version of rubocop
  we were using.

bwebster commented:
ping @brandonc @mhuggins.

The goal with this is to be able to graph our deployments in datadog, as well as overlay those events onto other charts.

mhuggins commented on lib/kapost_deploy/seismograph/notifier.rb:
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module KapostDeploy
+  module Seismograph
```
This class is cool, I like the way you're delegating with the null adaptor.

mhuggins reviewed (approved):
Very cool!

bwebster committed:
Reordered dependencies in gemspec.

- Appease codeclimate warning around listing dependencies in
  alphabetical order.
